### DISCUSSION TITLE
unable to install on ubuntu 18.04

### DISCUSSION THREAD

**Discussion by aludman** [ifm/ifm3d#430] [General]
Hi,

I'm unable to install [ifm3d-1.4.3](https://github.com/ifm/ifm3d/releases/download/v1.4.3/ifm3d-ubuntu-18.04-amd64-debs_1.4.3.tar) on ubuntu 18.04 due to some dependency issues.

`sudo dpkg -i ifm3d_1.4.3_amd64-common.deb`
> Selecting previously unselected package ifm3d-common.
> (Reading database ... 53193 files and directories currently installed.)
> Preparing to unpack ifm3d_1.4.3_amd64-common.deb ...
> Unpacking ifm3d-common (1.4.3) ...
> dpkg: dependency problems prevent configuration of ifm3d-common:
>  ifm3d-common depends on libstdc++6 (>= 8.4.0-1ubuntu1~18.04); however:
>   Version of libstdc++6:amd64 on system is 8-20180414-1ubuntu2.
> 
> dpkg: error processing package ifm3d-common (--install):
>  dependency problems - leaving unconfigured
> Errors were encountered while processing:
>  ifm3d-common

The released version of libstdc++6 is **_8-20180414-1ubuntu2_**. The **_8.4.0-1ubuntu1_** version seems to be a proposed version that was not released according to [this](https://launchpad.net/ubuntu/bionic/+package/libstdc++6).

I have the same issue with libcurl4 when installing ifm3d-device. The package is looking for **_7.58.0-2ubuntu3.24_** but the released version **_7.58.0-2ubuntu3_**.

How can I get around these dependency issues and install ifm3d on ubuntu 18.04?

**Comment by graugans** [comment]
We build our Debian Packages on the official Ubuntu Docker containers. Have you tried to perform a `apt-get update && apt-get dist-upgrade` before installing the ifm3d?

**Reply by aludman** [reply]
Indeed, it installs just fine in a new LXD container. I'm guessing that the reason it works in the container and not my other system is that the container is based on Ubuntu 18.04.6 LTS while the system I'm having problems with is still on the originally Ubuntu 18.04 LTS.

It turns out that the `dist-upgrade` did not work because my /etc/apt/sources.list did not include 'ubuntu bionic-updates'.
I've fixed that and upgrading the system now. I will confirm on Monday whether that worked o not but it should.

Thanks for the help!

**Comment by aludman** [comment]
This issue was caused by the fact that my system was not able to update itself to latest point release of Ubuntu 18.04 (i.e. 18.04.06). Once I sorted that out, I was able to install ifm3d-1.4.3 on Ubuntu 18.04 without any issues.
